modules/installation-osp-converting-ignition-resources.adoc

```diff
@@ -0,0 +1,111 @@
+// Module included in the following assemblies:
+//
+// * installing/installing_openstack/installing-openstack-user.adoc
+
+[id="installation-osp-converting-ignition-resources_{context}"]
+= Preparing the bootstrap Ignition file
+
+The bootstrap Ignition file that the installation program generates is too large to pass to {rh-openstack} directly.
+
+First, upload the bootstrap Ignition configuration file. Then, create a small Ignition file ("shim") that downloads the generated ignition file to Nova as user data.
+
+.Prerequisites
+
+* The bootstrap Ignition file that the installation program generates (`bootstrap.ign`)
+* The infrastructure ID from the installation program's metadata file is set as an environment variable (`$INFRA_ID`)
+** If the variable is not set, see *Creating the Kubernetes manifest and Ignition config files*.
+* An HTTP(S)-accessible way to store the bootstrap ignition file
+** The documented procedure uses Object Storage (Swift), but you could also use Amazon S3, an internal HTTP server, or an ad hoc Nova server
+
+.Procedure
+
+. Edit the bootstrap Ignition file to update the hostname file when it runs. You can do this manually, but it is faster to run the following Python script:
++
+[source,python]
+----
+import base64
+import json
+import os
+
+with open('bootstrap.ign', 'r') as f:
+    ignition = json.load(f)
+
+files = ignition['storage'].get('files', [])
+
+infra_id = os.environ.get('INFRA_ID', 'openshift').encode()
+hostname_b64 = base64.standard_b64encode(infra_id + b'-bootstrap\n').decode().strip()
+files.append(
+{
+    'path': '/etc/hostname',
+    'mode': 420,
+    'contents': {
+        'source': 'data:text/plain;charset=utf-8;base64,' + hostname_b64,
+        'verification': {}
+    },
+    'filesystem': 'root',
+})
+
+ignition['storage']['files'] = files;
+
+with open('bootstrap.ign', 'w') as f:
+    json.dump(ignition, f)
+----
+
+. Using the Swift CLI, create a container:
++
+----
+$ swift post <container_name>
+----
+
+. Upload the bootstrap Ignition file to the container:
++
+----
+$ swift upload <container_name> bootstrap.ign
+----
+
+. Set the container to be read-accessible:
++
+----
+$ swift post <container_name> --read-acl ".r:*,.rlistings"
+----
+
+. Retrieve `StorageURL`:
++
+----
+$ swift stat -v
+----
+** The URL should follow this format: `<StorageURL>/<container_name>/bootstrap.ign`
++
+[WARNING]
+The bootstrap Ignition file contains sensitive information, like clouds.yaml credentials and TLS certificates. Ensure that you store it in a secure place, and delete it after you complete the installation process.
```

Comment:
The TLS certificate included in the Ignition file is just the public part of it; it's therefore not sensitive information (however, clouds.yaml definitely is, exactly like you state)